test/src/e2e_vm_tests/mod.rs

```diff
@@ -218,12 +219,24 @@ fn discover_test_configs() -> Result<Vec<TestDescription>, String> {
     Ok(configs)
 }
 
+const DIRECTIVE_RX: &str = r"\b#\s*(\w+):\s+(.*)";
+
 fn parse_test_toml(path: &Path) -> Result<TestDescription, String> {
     let (toml_content, checker) = std::fs::read_to_string(path)
         .map_err(|e| e.to_string())
         .and_then(|toml_content_str| {
-            // Parse the file for FileCheck directives and_then parse the file into TOML.
-            filecheck::CheckerBuilder::new()
+            let mut checker = filecheck::CheckerBuilder::new();
+
+            // Parse the file and check for unknown FileCheck directives.
+            let re = Regex::new(DIRECTIVE_RX).unwrap();
+            for cap in re.captures_iter(&toml_content_str) {
+                if let Ok(false) = checker.directive(&cap[0]) {
+                    return Err(format!("Unknown FileCheck directive: {}", &cap[1]));
+                }
+            }
+
+            // Then parse the file into TOML.
+            checker
                 .text(&toml_content_str)
                 .map_err(|e| e.to_string())
                 .and_then(|checker| {
```

Comment:
GitHub won't let me highlight bits that aren't part of the diff, but you don't need to do this any more if you're adding the directives individually.

You only need to keep this:
```rust
                    toml_content_str
                        .parse::<toml::Value>()
                        .map_err(|e| e.to_string())
                        .map(|toml_content| (toml_content, checker.finish()))
```

In fact it could probably be simpilified some more.  It was originally a pipeline, hence all the `and_then`s etc, but now that `checker` is `mut` and built manually it can probably be done separately to parsing from TOML.